### PR TITLE
SSR to render materials properly for SEO (for unit pages).

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: src/components/Pages/WithPlayer/widgets/UnitMaterials/Summary/Summary.js [hunk 1]
```diff
@@ -12,35 +12,40 @@ import { selectors as assetsSelectors, doc2html } from '../../../../../../redux/
...
+      .map(f => f.language)) || [];
+
+export const getFile = (unit, lang) => {
+  if (!unit || !Array.isArray(unit.files)) {
+    return null;
+  }
+
+  return unit.files?.filter(f => f.language === lang)
+    .filter(f => MediaHelper.IsText(f) && !MediaHelper.IsPDF(f))
+    .find(f => f.insert_type === INSERT_TYPE_SUMMARY);
```
Comment (gshilin) [inline src/components/Pages/WithPlayer/widgets/UnitMaterials/Summary/Summary.js:28]: These 2 lines are the same as 17-18. Maybe to make it a function with a meaningful name?

File: src/components/Pages/WithPlayer/Playlist/PlaylistHeader.js [hunk 2]
```diff
@@ -12,20 +13,27 @@ import PlaylistPlayIcon from '../../../../images/icons/PlaylistPlay';
...
+  const unit                = useSelector(state => mdb.getDenormContentUnit(state.mdb, cuId || paramsId));
+  const c                   = canonicalCollection(unit);
+  const collection          = useSelector(state => mdb.getDenormCollection(state.mdb, cId || (c && c.id) || paramsId));
   const getPath             = useSelector(state => sources.getPathByID(state.sources));
 
   const { content_type, number, film_date, start_date, end_date, tag_id, source_id, likutim_id } = collection || false;
 
   const isLesson = COLLECTION_DAILY_LESSONS.includes(content_type);
 
+  if (!unit) {
```
Comment (gshilin) [inline src/components/Pages/WithPlayer/Playlist/PlaylistHeader.js:33]: Я бы это перенес до 29 строки. Чем раньше вернешься, тем меньше работы.

File: src/components/Pages/WithPlayer/widgets/UnitMaterials/Summary/Summary.js (hunk 1 above)
Comment (bbfsdev) [reply src/components/Pages/WithPlayer/widgets/UnitMaterials/Summary/Summary.js:28]: They are not the same.

First will "getAllSummaryFiles" and the one you commented on will "getFirstSummaryFileOrNull".
I tried refactoring this into "getAllTextFiles" and then getting the first or filtering summary (again) which does not result in a nice code... so I think we should keep this as is.

File: src/components/Pages/WithPlayer/Playlist/PlaylistHeader.js (hunk 2 above)
Comment (bbfsdev) [reply src/components/Pages/WithPlayer/Playlist/PlaylistHeader.js:33]: Good catch, done.

**Comment by gshilin** [comment]
lgtm

**Review by gshilin — APPROVED** [review]

**Pull request merged**